[Test/Sink] Add rosbag-based pipeline test cases for videotestsrc/audiotestsrc

wooksong opened the pull request:
Related epic: https://github.com/nnsuite/nnstreamer/issues/1059
See also: https://github.com/nnsuite/nnstreamer/issues/1121

This PR mainly concerns adding a set of utilities and test cases for rosbag-based pipeline test.

### Details ###
- [Test/Common] Add a python script for rosbag-based tests
  - ~~In the current version, comparing a raw data file containing video/x-raw with a rosbag file corresponding to the raw file is only supported. (it will be extended.)~~
  - Comparing a raw data file containing video/x-raw or audio/x-raw with a rosbag file corresponding to the raw file is supported now.
- 1) [Test/Sink] Add rosbag-based pipeline test cases
- 2) [Test/Sink] Add test cases covering audio format 
  - Raw data files created by dumping a video/x-raw frame (RGB, BGRx, and GRAY8) and 8000 audio/x-raw frames (S16LE, F32LE, and F64LE) using filesink are compared with rosbag files corresponding those raw data files.

### Changes in this PR ###
- [V2] Update test_utils.py to support audio/x-raw format
- [V2] Add est cases covering audio format 

### Test Results ###
```bash
./nnstreamer-ros/tests  (add-unittest-sink) $ ssat

==================================================

==================================================
    Test Group rosbag-based pipeline test Starts.
[PASSED] 1-1:gst-launch of case 1-1
[PASSED] 1-1:tensor_ros_sink: simple test case using raw data (video/BGRx/uint8)
[PASSED] 1-2:gst-launch of case 1-2
[PASSED] 1-2:tensor_ros_sink: simple test case using raw data (video/RGB/uint8)
[PASSED] 1-3:gst-launch of case 1-3
[PASSED] 1-3:tensor_ros_sink: simple test case using raw data (video/GRAY8/uint8)
[PASSED] 2-1:gst-launch of case 2-1
[PASSED] 2-1:tensor_ros_sink: simple test case using raw data (audio/mono/s16le/8k)
[PASSED] 2-2:gst-launch of case 2-2
[PASSED] 2-2:tensor_ros_sink: simple test case using raw data (audio/stereo/s16le/8k)
[PASSED] 2-3:gst-launch of case 2-3
[PASSED] 2-3:tensor_ros_sink: simple test case using raw data (audio/mono/f32le/8k)
[PASSED] 2-4:gst-launch of case 2-4
[PASSED] 2-4:tensor_ros_sink: simple test case using raw data (audio/stereo/f32le/8k)
[PASSED] 2-5:gst-launch of case 2-5
[PASSED] 2-5:tensor_ros_sink: simple test case using raw data (audio/mono/f64le/8k)
[PASSED] 2-6:gst-launch of case 2-6
[PASSED] 2-6:tensor_ros_sink: simple test case using raw data (audio/stereo/f64le/8k)
==================================================
[PASSED] Test Group rosbag-based pipeline test Passed

==================================================

[PASSED] tensor_ros_sink (18 passed among 18 cases)
==================================================
[PASSED] All Test Groups (1) Passed!
         TC Passed: 18 / Failed: 0
```

Signed-off-by: Wook Song <wook16.song@samsung.com>

wooksong committed:
[Test/Common] Add a python script for rosbag-based tests

This patch adds a python script, test_utils.py, for rosbag-based
pipeline test cases. In the current version, comparing a raw data file
containing video/x-raw with a rosbag file corresponding to the raw file
is only supported.

Signed-off-by: Wook Song <wook16.song@samsung.com>

wooksong committed:
[Test/Sink] Add rosbag-based pipeline test cases

This patch adds several rosbag-based pipeline test cases for a video/x-raw
stream whose types of BGRx, RGB, and GRAY8.

Signed-off-by: Wook Song <wook16.song@samsung.com>

myungjoo commented on tests/tensor_ros_sink/runTest.sh:
```diff
@@ -0,0 +1,36 @@
+
+PATH_TO_PLUGIN="../../build"
+
+gst-launch-1.0 -m -v --gst-plugin-path=${PATH_TO_PLUGIN} videotestsrc num-buffers=1 ! video/x-raw,format=BGRx,width=280,height=40,framerate=0/1 ! tee name=t ! queue ! tensor_converter silent=TRUE ! tensor_ros_sink dummy=true enable-save-rosbag=true location="simple.uint8.BGRx.bag" sync=true t. ! queue ! filesink location="simple.uint8.BGRx.log" sync=true
```
I recommend to use ```gstTest``` macro. It will report the result of gst-launch.

wooksong replied:
I will apply your comment. Thanks!

wooksong committed:
[DIST/Tizen/Test] Enable ssat test for rosbag-based pipeline test cases

This patch enables ssat test for rosbag-based pipeline test cases in
the Tizen rpm spec file.

Signed-off-by: Wook Song <wook16.song@samsung.com>

wooksong committed:
[DIST/Debian/Test] Enable rosbag-based pipeline test cases using SSAT

This patch enalbes rosbag-based pipeline test cases using SSAT for
Debian package build.

Signed-off-by: Wook Song <wook16.song@samsung.com>

myungjoo reviewed (approved):
LGTM!

wooksong committed:
[Test/Sink] Add test cases covering audio format

This patch adds test cases covering auido/x-raw format. In addition, the
python script for test utilities is modified to support audio/x-raw
format as well.

Signed-off-by: Wook Song <wook16.song@samsung.com>

again4you reviewed (approved):
LGTM!
